### PR TITLE
always use map to search for fields in client

### PR DESCRIPTION
**Pull request by 31453**

### CODE REVIEW

File: viewer/vueapp/src/store.js [hunk 1]
```diff
@@ -217,10 +218,24 @@ const store = new Vuex.Store({
...
       state.fieldhistory = value.fieldhistory.fields || [];
       state.esCluster.availableCluster = value.clusters;
+
+      // fieldsMap has keys for these fields: dbField, dbField2, fieldECS, and exp (id/key)
+      // fieldsAliasMap has keys for field aliases
+      for (const key in value.fieldsMap) {
+        const field = value.fieldsMap[key];
+        state.fieldsMap[field.exp] = field;
+        state.fieldsMap[field.dbField] = field;
+        state.fieldsMap[field.dbField2] = field;
```
Comment (awick) [inline viewer/vueapp/src/store.js:231]: dbField2 can be undefined if you want to check for it

File: viewer/vueapp/tests/consts.js [hunk 2]
```diff
@@ -519,4 +519,6 @@ export const fields = [{
 export const fieldsMap = {};
 for (const field of fields) {
   fieldsMap[field.exp] = field;
+  fieldsMap[field.dbField] = field;
+  fieldsMap[field.dbField2] = field;
```
Comment (awick) [inline viewer/vueapp/tests/consts.js:523]: dbField2 can be undefined, might not matter for tests

**Review by awick — APPROVED** [review]
seems to work :)

File: viewer/vueapp/src/store.js (hunk 1 above)
Comment (31453) [reply viewer/vueapp/src/store.js:231]: Got it, should I also fix this for `viewer/config.js`?
```
internals.fieldsMap[field._id] = source;
internals.dbFieldsMap[source.dbField] = source;
internals.dbFieldsMap[source.dbField2] = source;
```
(in the `loadFields` function)

Comment (awick) [reply viewer/vueapp/src/store.js:231]: sure!

File: viewer/vueapp/tests/consts.js (hunk 2 above)
Comment (31453) [reply viewer/vueapp/tests/consts.js:523]: I always define dbField2 in the tests, so this should be fine

File: viewer/vueapp/src/store.js (hunk 1 above)
Comment (31453) [reply viewer/vueapp/src/store.js]: 👍

**Pull request merged**